backend/experiment/actions/form.py

```diff
@@ -38,91 +62,280 @@ def __init__(
+    """A question that accepts text input.
+
+    Args:
+        input_type (str): Type of text input (default: 'text')
+        max_length (int): Maximum character length
+        **kwargs: Additional Question arguments
+    """
+
+    def __init__(self, input_type: str = "text", max_length: int = 64, **kwargs) -> None:
         super().__init__(**kwargs)
-        self.max_length = max_length # the maximum length of the question's answer in characters
+        self.max_length = max_length  # the maximum length of the question's answer in characters
         self.input_type = input_type
-        self.view = 'STRING'
+        self.view = "STRING"
 
 
 class BooleanQuestion(Question):
-    def __init__(self, choices=None, **kwargs):
+    """A yes/no question component.
+
+    Args:
+        choices (Optional[Dict[str, str]]): Custom yes/no labels
+        **kwargs: Additional Question arguments
+
+    Example:
+        ```python
+        question = BooleanQuestion(
+            key="is_student",
+            question="Are you a student?",
+            choices={
+                "no": "No", # or _("No") for translation
+                "yes": "Yes"
+            },
+        )
+        ```
+    """
+
+    def __init__(self, choices: Optional[Dict[str, str]] = None, **kwargs) -> None:
         super().__init__(**kwargs)
-        self.choices = choices or {
-            'no': _('No'),
-            'yes': _('Yes')
-        }
-        self.view = 'BUTTON_ARRAY'
-        self.style = {STYLE_BOOLEAN_NEGATIVE_FIRST: True, 'buttons-large-gap': True}
+        self.choices = choices or {"no": _("No"), "yes": _("Yes")}
+        self.view = "BUTTON_ARRAY"
+        self.style = {STYLE_BOOLEAN_NEGATIVE_FIRST: True, "buttons-large-gap": True}
 
 
 class ChoiceQuestion(Question):
-    def __init__(self, choices, min_values=1, **kwargs):
+    """A question with (multiple) choice options.
+
+    Args:
+        choices (Dict[str, str]): Available options
+        min_values (int): Minimum number of selections required
+        **kwargs: Additional Question arguments
+
+    Example:
+        ```python
+        question = ChoiceQuestion(
+            key="color",
+            question="What's your favorite color?",
+            choices={
+                "red": "Red", # or _("Red") for translation
+                "green": "Green",
+                "blue": "Blue",
+            },
+            min_values=1,
+        )
+        ```
+
+    Note:
+        - To have multiple choices (participant can select more than one answer), set `view` to 'CHECKBOXES'
+    """
+
+    def __init__(self, choices: Dict[str, str], min_values: int = 1, **kwargs) -> None:
         super().__init__(**kwargs)
         self.choices = choices
         # minimal number of values to be selected, 1 or more
+        # TODO: enforce (raise ValueError), or math.floor it
         self.min_values = min_values
 
 
 class DropdownQuestion(Question):
```

Comment:
Should these all inherit `ChoiceQuestion`, rather than plan `Question`? `BooleanQuestion` always comes with yes/no buttons at this point.